usaspending_api/search/v2/views/search.py

```diff
@@ -143,253 +143,19 @@ class SpendingByCategoryVisualizationViewSet(APIView):
-                if can_use_view(filters, 'SummaryPscCodesView'):
-                    queryset = get_view_queryset(filters, 'SummaryPscCodesView')
-                    queryset = queryset \
-                        .filter(product_or_service_code__isnull=False) \
-                        .values(psc_code=F("product_or_service_code"))
-                else:
-                    queryset = queryset \
-                        .filter(psc_code__isnull=False) \
-                        .values("psc_code")
-
-                queryset = sum_transaction_amount(queryset, 'aggregated_amount', filter_types=filter_types) \
-                    .order_by('-aggregated_amount')
-                # Begin DB hits here
-                results = list(queryset[lower_limit:upper_limit + 1])
-
-                page_metadata = get_simple_pagination_metadata(len(results), limit, page)
-                results = results[:limit]
-
-                response = {"category": category, "scope": scope, "limit": limit, "results": results,
-                            "page_metadata": page_metadata}
-                return Response(response)
-
-            elif scope == "naics":
-                if can_use_view(filters, 'SummaryNaicsCodesView'):
-                    queryset = get_view_queryset(filters, 'SummaryNaicsCodesView')
-                    queryset = queryset \
-                        .filter(naics_code__isnull=False) \
-                        .values('naics_code')
-                    queryset = sum_transaction_amount(queryset, 'aggregated_amount', filter_types=filter_types) \
-                        .order_by('-aggregated_amount') \
-                        .values(
-                            'naics_code',
-                            'aggregated_amount',
-                            'naics_description')
-                else:
-                    queryset = queryset \
-                        .filter(naics_code__isnull=False) \
-                        .values("naics_code")
-                    queryset = sum_transaction_amount(queryset, 'aggregated_amount', filter_types=filter_types) \
-                        .order_by('-aggregated_amount') \
-                        .values(
-                            'naics_code',
-                            'aggregated_amount',
-                            'naics_description')
-
-                # Begin DB hits here
-                results = list(queryset[lower_limit:upper_limit + 1])
-
-                page_metadata = get_simple_pagination_metadata(len(results), limit, page)
-                results = results[:limit]
-
-                response = {"category": category, "scope": scope, "limit": limit, "results": results,
-                            "page_metadata": page_metadata}
-                return Response(response)
-
-            else:  # recipient_type
-                raise InvalidParameterException("recipient type is not yet implemented")
+        return Response(spending_by_category_logic(validated_payload))
```

Comment:
Passing an class's attribute to the Response could clarify more what is actual returned in the response:
```spending_by_category_logic(validated_payload).results```